### ISSUE TITLE
Wherigo: Error on loading savegame

### ISSUE THREAD

**Issue by rouroni1** [cgeo/cgeo#16313]
### Describe your problem!

Error message from OpenWIG when loading a wherigo savegame from:
GC5ARN2 ("Uma Aventura no Carrical" (GC5ARN2, cguid: c1a4fd95-07d7-4667-94c1-d2e6eead9cae)
GC5PA6D

### How to reproduce?

1. Start the wherigo cartridge
2. Savegame in any slot, and stop cartridge
3. Start the same cartridge and load the savegame

### Actual result after these steps?

Get a message with "Wherigo error", but the savegame is loaded, if I it "cancel" I can resume the game.

![Screenshot_2024-11-02-21-14-02-930_cgeo geocaching](https://github.com/user-attachments/assets/82f806dd-849d-42d8-a88b-ee7cee1273e2)

### Reproducible

Yes

### c:geo Version

2024.11.02-NB-1da1599

**Comment by rouroni1** [comment]
![Screenshot_2024-11-02-21-14-02-930_cgeo geocaching](https://github.com/user-attachments/assets/82f806dd-849d-42d8-a88b-ee7cee1273e2)

**Comment by eddiemuc** [comment]
Confirmed. Stacktrace for GC5ARN2:

```
2024-11-03 15:16:49.308 13675-14019                             I  [Thread-4] WHERIGO: load save game: Savegame: 1-GC5ARN2/File:name='c1a4fd95-07d7-4667-94c1-d2e6eead9cae_uma_aventura_no_ca-1-GC5ARN2.sav', uri=content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fwherigo%2Fc1a4fd95-07d7-4667-94c1-d2e6eead9cae_uma_aventura_no_ca-1-GC5ARN2.sav, isDirectory=false, dirLocation=null, mimeType=application/octet-stream
2024-11-03 15:16:49.338 13675-14019 System.err                  W  java.lang.RuntimeException: attempted index of non-table
2024-11-03 15:16:49.339 13675-14019                             W  	at se.krka.kahlua.stdlib.BaseLib.fail(BaseLib.java:390)
2024-11-03 15:16:49.340 13675-14019                             W  	at se.krka.kahlua.stdlib.BaseLib.luaAssert(BaseLib.java:385)
2024-11-03 15:16:49.340 13675-14019                             W  	at se.krka.kahlua.vm.LuaState.tableSet(LuaState.java:1213)
2024-11-03 15:16:49.340 13675-14019                             W  	at se.krka.kahlua.vm.LuaState.luaMainloop(LuaState.java:359)
2024-11-03 15:16:49.340 13675-14019                             W  	at se.krka.kahlua.vm.LuaState.call(LuaState.java:217)
2024-11-03 15:16:49.340 13675-14019                             W  	at se.krka.kahlua.vm.LuaState.call(LuaState.java:1136)
2024-11-03 15:16:49.341 13675-14019                             W  	at cz.matejcik.openwig.EventTable.callEvent(EventTable.java:112)
2024-11-03 15:16:49.342 13675-14019                             W  	at cz.matejcik.openwig.Task.setItem(Task.java:29)
2024-11-03 15:16:49.342 13675-14019                             W  	at cz.matejcik.openwig.EventTable.rawset(EventTable.java:131)
2024-11-03 15:16:49.342 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.deserializeLuaTable(Savegame.java:340)
2024-11-03 15:16:49.342 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreObject(Savegame.java:284)
2024-11-03 15:16:49.344 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreValue(Savegame.java:263)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.EventTable.deserialize(EventTable.java:39)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreObject(Savegame.java:305)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreValue(Savegame.java:263)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.deserializeLuaTable(Savegame.java:338)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreObject(Savegame.java:284)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreValue(Savegame.java:263)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.deserializeLuaTable(Savegame.java:338)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreObject(Savegame.java:284)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreValue(Savegame.java:263)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.EventTable.deserialize(EventTable.java:39)
2024-11-03 15:16:49.348 13675-14019                             W  	at cz.matejcik.openwig.Cartridge.deserialize(Cartridge.java:116)
2024-11-03 15:16:49.349 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreObject(Savegame.java:305)
2024-11-03 15:16:49.349 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restoreValue(Savegame.java:263)
2024-11-03 15:16:49.349 13675-14019                             W  	at cz.matejcik.openwig.formats.Savegame.restore(Savegame.java:89)
2024-11-03 15:16:49.349 13675-14019                             W  	at cz.matejcik.openwig.Engine.restoreGame(Engine.java:149)
2024-11-03 15:16:49.349 13675-14019                             W  	at cz.matejcik.openwig.Engine.run(Engine.java:210)
2024-11-03 15:16:49.349 13675-14019                             W  	at java.lang.Thread.run(Thread.java:1012)
2024-11-03 15:16:49.349 13675-14019 System.out                  I  at @E:/webtempfiles/7942b669-ac41-4bda-a516-1113979f1fd5/_cartridge-new.enc:1136
2024-11-03 15:16:49.349 13675-14019 cgeo                        W  [Thread-4] WHERIGOGAME: ERROR: you hit a bug! please report at openwig.googlecode.com and i'll fix it for you!
                                                                   java.lang.RuntimeException: attempted index of non-table
                                 
```

**Comment by eddiemuc** [comment]
I only tried GC5ARN2

This seems to be a bug in the cartrdige. Same happens in WhereYouGo, but WhereYouGo "hides" the bug first (I believe by an activity overlapping the error dialog). 

When you follow same steps as above in WhereYouGo, then after step 3 you don't get an error. But if you then immediately stop the game (by pressing "back" button two times), then the very same error will show up in a dialog.

I don't really believe we should hide such errors. But maybe more clearly mark them as errors in the cartridge (-> ask the player to contact the author)? 

NB: this cartridge also causes an interesting bug in WhereYouGo in that my phone does not stop playing the cartridge's song. Even after I completely ended WhereYouGo. But this cartridge also revealed a bug in c:geo wherigo player: the playback of the song is stopped after a while w/o reason

**Comment by rouroni1** [comment]
> I don't really believe we should hide such errors. But maybe more clearly mark them as errors in the cartridge (-> ask the player to contact the author)?
> 

I agree with this solution. 
I've investigated a little, both cartridges were made by the same owner / person.

For context: I don't recall to see the issue when I played the cartridge at the time, but now I was testing all the cartridges I've found and stumbled on the issue.

**Comment by rouroni1** [comment]
Fixed with nightly 2025.05.03-NB-a76041c .
